embed/ephy-fx-password-notification.c

```diff
@@ -0,0 +1,162 @@
+}
+
+static void
+ephy_fx_password_notification_dispose (GObject *object)
+{
+  EphyFxPasswordNotification *self = EPHY_FX_PASSWORD_NOTIFICATION (object);
+
+  g_clear_pointer (&self->user, g_free);
+
+  G_OBJECT_CLASS (ephy_fx_password_notification_parent_class)->dispose (object);
+}
+
+static void
+ephy_fx_password_notification_set_property (GObject      *object,
+                                            guint         prop_id,
+                                            const GValue *value,
+                                            GParamSpec   *pspec)
+{
+  EphyFxPasswordNotification *self = EPHY_FX_PASSWORD_NOTIFICATION (object);
+
+  switch (prop_id) {
+    case PROP_USER:
+      self->user = g_value_dup_string (value);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+ephy_fx_password_notification_get_property (GObject      *object,
+                                            guint         prop_id,
+                                            GValue       *value,
+                                            GParamSpec   *pspec)
+{
+  EphyFxPasswordNotification *self = EPHY_FX_PASSWORD_NOTIFICATION (object);
+
+  switch (prop_id) {
+    case PROP_USER:
+      g_value_set_string (value, self->user);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+ephy_fx_password_notification_init (EphyFxPasswordNotification *self)
+{
+  g_return_if_fail (EPHY_IS_FX_PASSWORD_NOTIFICATION (self));
+
+  self->note = gtk_label_new (NULL);
+  gtk_widget_set_halign (self->note, GTK_ALIGN_CENTER);
+  gtk_widget_set_hexpand (self->note, TRUE);
+  gtk_grid_attach (GTK_GRID (self), self->note, 0, 0, 1, 1);
+
+  self->suggestion = gtk_label_new (NULL);
+  gtk_widget_set_halign (self->suggestion, GTK_ALIGN_CENTER);
+  gtk_widget_set_hexpand (self->suggestion, TRUE);
+  gtk_grid_attach (GTK_GRID (self), self->suggestion, 0, 1, 1, 1);
+}
+
+static void
+ephy_fx_password_notification_class_init (EphyFxPasswordNotificationClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->constructed = ephy_fx_password_notification_constructed;
+  object_class->dispose = ephy_fx_password_notification_dispose;
+  object_class->set_property = ephy_fx_password_notification_set_property;
+  object_class->get_property = ephy_fx_password_notification_get_property;
+
+  g_object_class_install_property (object_class,
+                                   PROP_USER,
+                                   g_param_spec_string ("user",
+                                                        "User",
+                                                        "The email of the signed in user",
+                                                        "",
+                                                        G_PARAM_CONSTRUCT_ONLY | G_PARAM_WRITABLE));
```

Comment:
Also pass `G_PARAM_STATIC_STRINGS`, which you always want to pass to every property you create, unless you are doing something crazy. Otherwise, the strings you passed to this function will get duplicated and freed unnecessarily. That's not needed, as they're just string constants. The default behavior was a mistake, it envisioned applications changing the names or descriptions of properties at runtime, which is of course nuts.

(GObject also allows you to inherit from new interfaces at runtime. Can you imagine why that is a bad idea. :)